internal/venice-common/src/main/java/com/linkedin/venice/utils/PartitionUtils.java

```diff
@@ -23,41 +24,40 @@ public class PartitionUtils {
   private static final Logger LOGGER = LogManager.getLogger(PartitionUtils.class);
 
   /**
-   * Calculate partition count for new version. If the version is the first one of the given store,
-   * calculate the number by given store size and partition size. Otherwise, use the number from the current active
-   * version.
+   * Calculate partition count for new version. If store level partition count is not configured (0),
+   * calculate the number by storage quota and partition size. Otherwise, use the store level partition count.
    */
   public static int calculatePartitionCount(
       String storeName,
-      long storeSizeBytes,
-      int previousPartitionCount,
+      long storageQuota,
+      int storePartitionCount,
       long partitionSize,
       int minPartitionCount,
       int maxPartitionCount) {
-    if (storeSizeBytes <= 0) {
-      throw new VeniceException("Store size: " + storeSizeBytes + " is invalid.");
+    if (storageQuota <= 0 && storageQuota != Store.UNLIMITED_STORAGE_QUOTA) {
+      throw new VeniceException("Storage quota: " + storageQuota + " is invalid.");
     }
-    if (previousPartitionCount == 0) {
-      // First Version, calculate partition count
-      long partitionCount = storeSizeBytes / partitionSize;
+    if (storePartitionCount == 0) {
+      // Store level partition count is not configured, calculate partition count
+      long partitionCount = storageQuota / partitionSize;
```

Comment:
As discussed internally, let us scan the existing stores to see how big the impact will be with the new partition count calculation logic?
Essentially, it is good to know how many new partitions will be added because of this new logic.

Reply:
Also where do we set the calculated partition count on the store? Will this be recalculated on every push?

Reply:
Will do.

Reply:
Thanks

Reply:
After scanning existing stores, I find that the new calculation by storage quota logic will add about 11k partitions to prod, which is fine from SRE's point of view.